### PR TITLE
Client timer

### PR DESCRIPTION
**Pull request by jpfr**
Follow up for #1497.
Implements timed callbacks for the client.
Securechannel and session keepalive is intended to be moved over here.

### CODE REVIEW

**Comment by StalderT** [comment]
👍

**Comment by StalderT** [comment]
Can we add something like «getfd» to the connection? Then the client can use select to wait an event on the socket or on a external event. If you agree I can work on it.

**Comment by jpfr** [comment]
The connection struct already contains that.
https://github.com/open62541/open62541/blob/master/include/ua_plugin_network.h#L63

So this is not the preferred way to drive clients. But on systems with mutliple clients, it is advantageous to `select` on multiple sockets at once. Hence the connection is exposed to power-users.

**Comment by StalderT** [comment]
Thank you for your reply. I did not pay attention to the fact that the content of the structure was publicly available. Sorry!

File: src/client/ua_client.c [hunk 1]
```diff
@@ -354,11 +421,78 @@ __UA_Client_AsyncService(UA_Client *client, const void *request,
...
+                *nextTimeout = UA_MAXTIMEOUT;
+            else
+                *nextTimeout = (UA_UInt16)nextCall;
+        }
+    }
+
     return retval;
 }
+
+/* A worker thread could be spliced into here... */
```
Comment (sherylll) [inline src/client/ua_client.c]: Work thread already implemented in my fork. Will merge that

**Comment by Pro** [comment]
As discussed in the telco today, this code is completely outdated and partially included in master.

**Pull request closed without merging**